src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py

```diff
@@ -1050,7 +1048,7 @@ def test_vm_extension_with_id(self, resource_group):
         self.kwargs.update({
             'vm': 'vm1'
         })
-        vm_id = self.cmd('vm create -g {rg} -n {vm} --image UbuntuLTS').get_output_in_json()['id']
+        vm_id = self.cmd('vm create -g {rg} -n {vm} --image UbuntuLTS --generate-ssh-keys').get_output_in_json()['id']
```

Comment:
why will previous test fail without `--generate-ssh-keys` ?

Reply:
Yes. It can pass in local machine because local machine has a ssh key in user folder. But a clean environment in pipeline does not have one.